src/Compilers/CSharp/Portable/BoundTree/DecisionTree.cs

```diff
@@ -197,6 +198,11 @@ internal DecisionTree ComputeDecisionTree()
                         else
                         {
                             AddToDecisionTree(result, label);
+                            if (label.Guard == null || label.Guard.ConstantValue == ConstantValue.True)
+                            {
+                                // Only unconditional switch labels contribute to subsumption
+                                AddToDecisionTree(subsumptionTree, label);
```

Comment:
CheckSubsumped just looks in the subsumption tree to see if the proposed new pattern is already handled. It doesn't modify the tree.

---
In reply to: [79641466](https://github.com/dotnet/roslyn/pull/13770#discussion_r79641466) [](ancestors = 79641466,79474816)